packages/devtools/lib/src/vm_service_wrapper.dart

```diff
@@ -378,21 +406,22 @@ class VmServiceWrapper implements VmService {
   /// used after a hot restart to avoid bugs where we have zombie futures lying
   /// around causing tests to flake.
   void doNotWaitForPendingFuturesBeforeExit() {
-    allFuturesCompleted = Completer<bool>();
-    allFuturesCompleted.complete(true);
-    _activeFutures.clear();
+    _allFuturesCompleter = Completer<bool>();
+    _allFuturesCompleter.complete(true);
+    activeFutures.clear();
   }
 
-  Future<T> _trackFuture<T>(Future<T> future) {
-    if (allFuturesCompleted.isCompleted) {
-      allFuturesCompleted = Completer<bool>();
+  Future<T> _trackFuture<T>(String name, Future<T> future) {
+    final trackedFuture = new TrackedFuture(name, future);
```

Comment:
nit: keep a map from Future to debug description instead of creating the TrackedFuture class.
that will make it a little easier to toggle this on and off.

Reply:
I made this change, however it occurs to me that if we want to disable tracking wouldn't we be better just skipping the whole `trackFuture` call? As far as I can tell, it's only tests that ever use the tracked futures, so if we want to disable them for production I think it would be better to not track them at all, rather than track them and just don't keep their descriptions? (in which case, I think going back to `TrackedFuture` would be better than keeping two lists in-sync). WDYT?

Reply:
I've reverted the Map and left it as it was, but with a bool to opt-in to tracking only for tests. I think this is better (keeps the code clean instead of having a list and a map to track, while still avoiding any tracking in production). PTAL!